src/main/scala/tilelink/Nodes.scala

```diff
@@ -43,6 +43,13 @@ case class TLAdapterNode(
 
 case class TLIdentityNode()(implicit valName: ValName) extends IdentityNode(TLImp)()
 
+object TLNameNode {
+  def apply(name: ValName) = TLIdentityNode()(name)
+  def apply(name: String): TLIdentityNode = apply(Some(name))
+  def apply(name: Option[String]) =
+    TLIdentityNode()(ValName(name.getOrElse("with_no_name")))
+}
+
```

Comment:
This is pretty handy, but I wonder if it should exist for Interrupts and AXI as well?

Reply:
Yeah I wanted your approval of it first. Will do